### PR TITLE
JSON exporter

### PR DESCRIPTION
**Pull request by d-michail**
A simple JSON exporter. See #768.

----

- [x] I read and understood <https://github.com/jgrapht/jgrapht/wiki/Become-a-Contributor>
- [x] I read and understood <https://github.com/jgrapht/jgrapht/wiki/How-to-make-your-first-%28code%29-contribution>
- [x] I added [unit tests](https://github.com/jgrapht/jgrapht/wiki/Unit-testing)
- [x] I added [documentation](https://github.com/jgrapht/jgrapht/wiki/How-to-write-documentation)
- [x] I followed the [Coding and Style Conventions](https://github.com/jgrapht/jgrapht/wiki/Coding-and-Style-Conventions)
- [x] I **have not** modified `HISTORY.md` or `CONTRIBUTORS.md`
- [x] I ensured that [the git commit message is a good one](https://github.com/joelparkerhenderson/git_commit_message)

### CODE REVIEW

File: jgrapht-io/src/main/java/org/jgrapht/io/JSONExporter.java [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Objects;
+import java.util.Set;
+
+import org.apache.commons.text.StringEscapeUtils;
+import org.jgrapht.Graph;
+
+/**
+ * Exports a graph to a JSON format.
```
Comment (jkinable) [inline jgrapht-io/src/main/java/org/jgrapht/io/JSONExporter.java]: I'm not particularly well-versed in JSON, but is there like a standard which describes the format that we are exporting to? E.g. some RFC? And are we compliant with a specific JSON version? If so then we should probably mention these details here.

Comment (d-michail) [reply jgrapht-io/src/main/java/org/jgrapht/io/JSONExporter.java]: Well, JSON is so common these days that there is no need for a reference to a document describing it. I can probably change the name to a href pointing to json.org.

If you mean about the exact schema that we are using here, no there is no standard for exporting graphs. I used a very simple one which is a list of nodes and a list of edges, which is compatible with the one used by the sigmajs Javascript library which contains graph drawing algorithms.

I will try improve the javadocs slightly.

**Comment by d-michail** [comment]
Improved the javadocs, added a link to the RFC and added type support.

File: jgrapht-io/src/main/java/org/jgrapht/io/JSONExporter.java [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+        out.print(quoted(VERSION));
+
+        /*
+         * Vertices
+         */
+        out.print(',');
+        out.print(quoted("nodes"));
+        out.print(':');
+        out.print('[');
+        int j = 0;
```
[the remark this answers is not in the document]
Comment (jkinable) [reply jgrapht-io/src/main/java/org/jgrapht/io/JSONExporter.java]: Could you fix this one as well?

[the remark this answers is not in the document]
Comment (d-michail) [reply jgrapht-io/src/main/java/org/jgrapht/io/JSONExporter.java]: Sorry, I missed your point. What should I fix?

[the remark this answers is not in the document]
Comment (d-michail) [reply jgrapht-io/src/main/java/org/jgrapht/io/JSONExporter.java]: Aaah, now I see your point. Will do.

**Comment by jkinable** [comment]
great, looks good. thanks!

**Pull request merged**